### PR TITLE
Initial upload

### PR DESCRIPTION
**Pull request by vincerubinetti**
Over the past few months I've been growing increasingly frustrated with configuring eslint, for a couple of reasons:

1) I often have to dig through the dependency config files (the rule sets we extend from) to find out what a particular rule is set to. 
2) Some of the dependency config files are a bit outdated, like the google one. 
3) The overrides that I've put into my Hetmech and Adage eslint config files are becoming large and out of sync with one-another. It'd be good to have a centralized, consistent config file. 
4) It is a pain to have to install all these dev dependencies all the time, like `eslint-config-google`, `eslint-plugin-html`, etc. It'd be nice to simply install just `eslint-config-greenelab`, then make my `.eslintrc.yaml` simply be `extends: eslint-config-greenelab` and be completely done with it.
5) I'm tired of dealing with all these different/separate config files, each with their own format, filename, order of rules, etc. It'd be nice to only have to worry about one.

This PR attempts to alleviate those concerns.

The config file in this repo looks a bit different from most eslint config files in a few ways that I think have benefits. I don't "import" (extend from) anything. I have copied ALL of the rules from their respective pages, in the same order as they appear. This prevents confusion about rules that override each other, and what order other config files are imported in. Nothing is imported. Nothing is commented out. ALL of the possible rules (given the set of plugins we're using) are there, explicitly set to `off`, `error`, or `warning`. **It is declarative, explicit, and a single source of truth.**

For the `core` (ESLint built-in) rules, I've started with the google config, and made a few changes. For the most part, if a rule was on in the google config, it is on in this one. And then some additional rules are turned on that are general good practice. So it is a little more strict than google.

For the `react` and `jsx-a11y` plugins, I've simply set the recommend ones on and the others off. It's true that doing it the way I have done it here decouples us from the React recommended rules, if they ever change what they recommend. But I feel these are things we should be making opinionated choices about anyway.

Even if we decide we want to still extend from some configs (though I highly recommend against extending the google one), we should still make this a centralized package that installs all the dependencies for you. I also feel that specifying it in `yaml` makes it a lot cleaner.

### CODE REVIEW

**Comment by vincerubinetti** [comment]
As for reviewing this, it's probably not going to be possible to give a full review. It's more something I'm just going to have to try out for a while, and see if it causes any issues, and update the rules appropriately.

**Comment by vincerubinetti** [comment]
Regarding my dream to just be using `extends: eslint-config-greenelab` and nothing else, I may have to change `peerDependencies` to `devDependencies`. I'm not sure. I want whatever is necessary such that installing just `eslint-config-greenelab` also installs all the necessary plugins/packages/configs.

**Comment by vincerubinetti** [comment]
One other concern: `prettier`.

ESLint's autofix is great for almost everything. But one thing it can't do is auto-wrap lines that are too long (eg past 80 characters wide). But that is something that Prettier CAN do.

So my current workflow is to run prettier first, then eslint. It would be nice to incorporate the prettier step into this config, if possible, so two tools and commands aren't necessary. 

I don't know how easy this would be, given that there is a lot of feature overlap between prettier and eslint. I would really only want to use prettier for the auto-wrap capability, because everything else can be handled by eslint.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+yarn.lock
```
Comment (dongbohu) [inline .gitignore]: New line character needed.

**Pull request merged**